### ISSUE TITLE
Adding lenses to the mix

### ISSUE THREAD

**Issue by lbreede** [AntonPalmqvist/physically-based-api#18]
I've dabbled in physical lens emulation in Houdini and I wonder if there's a standard for lens characteristics like distortion, vignetting etc.
USD wise, it can probably be a focal length override on the camera, but I imagine the simulation is different from renderer to renderer. I've only ever tried the Karma physical lens shader. Just food for thought :)

**Comment by AntonPalmqvist** [comment]
Unfortunately there doesn't seem to be any standard just yet. As you say, it varies from renderer to renderer what kind of lens characteristics you can use.

Blender, for example, uses [4th degree polynomials for lens distortion](https://docs.blender.org/manual/en/3.1/render/cycles/object_settings/cameras.html#fisheye-lens-polynomial), while the data you get from databases such as [Lensfun](https://github.com/lensfun/lensfun) is not compatible with that format.

Other interesting links on the subject. However, none of them seem to have converged around a standard:
- http://zenopelgrims.com/project-polynomial-optics-arnold.html
- https://docs.chaos.com/display/VMAYA/Lens+analysis+tool
- https://www.panotools.org/dersch/barrel/barrel.html
- https://www.lentil.xyz/index.html#features
- https://github.com/hanatos/lensoptics

I'm hoping to expand the database with this kind of data, so please let me know if you have additional resources on this topic!

**Comment by lbreede** [comment]
yeah I'd usually point to Zeno's Lentil work. It's the best. The [Karma Physical Lens Shader](https://www.sidefx.com/docs/houdini/solaris/karma_lens_shader.html) is getting better and better and runs on the GPU finally :)

**Comment by lbreede** [comment]
If you look at [my fork](https://github.com/lbreede/physically-based-api/actions/workflows/tests.yml) you can see that there were tests that ran for every push and PR. Number 5 failed for example because changed a sensor size to be a list of three numbers. Therefore: 
`test_sensorSizes
    self.assertEqual(len(size), 2)
AssertionError: 3 != 2`
